Maskbook/Extension/UIView+Shake.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  UIView+Shake.swift
+//  Maskbook
+//
+//  Created by xiaojian sun on 2022/2/23.
+//  Copyright © 2022 dimension. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+enum ShakeDirection {
+    case horizontal
+    case vertical
+}
+// swiftlint:disable function_parameter_count legacy_random multiline_arguments
+extension UIView {
+    private func degreesToRadians(_ x: CGFloat) -> CGFloat {
+        .pi * x / 180.0
+    }
+
+    func shakeWhenDelete(duration: Double = 0.25,
+                         displacement: CGFloat = 1.0,
+                         degreesRotation: CGFloat = 2.0)
+    {
+        let negativeDisplacement = -1.0 * displacement
+        let position = CAKeyframeAnimation(keyPath: "position")
+        position.beginTime = 0.8
+        position.duration = duration
+        position.values = [
+            NSValue(cgPoint: CGPoint(x: negativeDisplacement, y: negativeDisplacement)),
+            NSValue(cgPoint: CGPoint(x: 0, y: 0)),
+            NSValue(cgPoint: CGPoint(x: negativeDisplacement, y: 0)),
+            NSValue(cgPoint: CGPoint(x: 0, y: negativeDisplacement)),
+            NSValue(cgPoint: CGPoint(x: negativeDisplacement, y: negativeDisplacement))
+        ]
+        position.calculationMode = CAAnimationCalculationMode.linear
+        position.isRemovedOnCompletion = false
+        position.repeatCount = Float.greatestFiniteMagnitude
+        position.beginTime = CFTimeInterval(Float(arc4random()).truncatingRemainder(dividingBy: Float(25)) / Float(100))
+        position.isAdditive = true
+
+        let transform = CAKeyframeAnimation(keyPath: "transform")
+        transform.beginTime = 2.6
+        transform.duration = duration
+        transform.valueFunction = CAValueFunction(name: CAValueFunctionName.rotateZ)
+        transform.values = [
+            degreesToRadians(-1.0 * degreesRotation),
+            degreesToRadians(degreesRotation),
+            degreesToRadians(-1.0 * degreesRotation)
+        ]
+        transform.calculationMode = CAAnimationCalculationMode.linear
+        transform.isRemovedOnCompletion = false
+        transform.repeatCount = Float.greatestFiniteMagnitude
+        transform.isAdditive = true
+        transform.beginTime = CFTimeInterval(Float(arc4random()).truncatingRemainder(dividingBy: Float(25)) / Float(100))
+
+        layer.add(position, forKey: "shakePosition")
```

Comment:
`position` and `transform` can be combined with `CAAnimationGroup`.

Reply:
I think It is not necessary.
If the animation group  need  interact with other animation, That's necessary.